client/src/components/player/VolumeControls.js

```diff
@@ -32,23 +32,126 @@ const ProgressBar = styled.div`
   width: 100%;
 `;
 
-function VolumeControls() {
-  return (
-    <VolumeControlsWrapper>
-      <VolumeBar>
-        <PlayerButton title="Enable Repeat" color="white">
-          <i className="material-icons md-20">volume_up</i>
-        </PlayerButton>
-        <ProgressBar>
-          <PlayerPlaybackProgress>
-            <PlayerPlaybackProgressBg>
-              <PlayerPlaybackProgressBgCharge progress={'7'} />
-            </PlayerPlaybackProgressBg>
-          </PlayerPlaybackProgress>
-        </ProgressBar>
-      </VolumeBar>
-    </VolumeControlsWrapper>
-  );
+class VolumeControls extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      volume: {
+        mouseDown: false,
+        current: 20
+      }
+    };
+
+    this.volumeRef = React.createRef();
+  }
+
+  componentDidMount = () => {
+    const { audioRef } = this.props;
+
+    audioRef.current.volume = 0.2;
+    this.updateVolume();
+    this.setMouseDownDefault();
+  };
+
+  updateVolume = () => {
+    const { audioRef } = this.props;
+
+    audioRef.current.addEventListener('volumechange', () => {
+      const volume = audioRef.current.volume * 100;
+
+      this.setState(prevState => ({
+        volume: { ...prevState.volume, current: volume }
+      }));
+    });
+  };
+
+  getVolumeBarPercentage = e =>
+    e.nativeEvent.offsetX / this.volumeRef.current.offsetWidth;
+
+  onClickChangeVolume = e => {
+    const percentage = this.getVolumeBarPercentage(e);
+
+    if (percentage >= 0 && percentage <= 1) {
+      this.setVolume(percentage);
+    }
+  };
+
+  onMouseMoveChangeVolume = e => {
+    const { volume } = this.state;
+
+    if (volume.mouseDown) {
+      const percentage = this.getVolumeBarPercentage(e);
+
+      if (percentage >= 0 && percentage <= 1) {
+        this.setVolume(percentage);
+      }
+    }
+  };
+
+  toggleMouseDownVolume = () =>
+    this.setState(prevState => ({
+      volume: {
+        ...prevState.volume,
+        mouseDown: !prevState.volume.mouseDown
+      }
+    }));
+
+  setVolume = volume => {
+    const { audioRef } = this.props;
+
+    audioRef.current.volume = volume;
+  };
+
+  onClickToggleVolume = () => {
+    const { audioRef } = this.props;
+
+    audioRef.current.muted = !audioRef.current.muted;
+  };
+
+  setMouseDownDefault = () => {
+    document.addEventListener('mouseup', () =>
```

Comment:
This is a bit tricky, it can cause closure in a lot of cases, making the code below live even if the component is destroyed, needs further investigation, also you need to remove this event listener